openmdao/core/system.py

```diff
@@ -4954,6 +4958,95 @@ def _get_vars_exec_order(self, inputs=False, outputs=False, variables=None, loca
 
         return var_list
 
+    def list_options(self,
+                     include_default=True,
+                     include_solvers=True,
+                     out_stream=_DEFAULT_OUT_STREAM,
+                     return_format='list'):
+        """
+        Write a list of output names and other optional information to a specified stream.
+
+        Parameters
+        ----------
+        include_default : bool
+            When True, include the built-in openmdao system options. Default is True.
+        include_solvers : bool
+            When True, include options from nonlinear_solver and linear_solver.
+        out_stream : file-like
+            Where to send human readable output. Default is sys.stdout.
+            Set to None to suppress.
+        return_format : str
+            Indicates the desired format of the return value. Can have value of 'list' or 'dict'.
+            If 'list', the return value is a list of (name, metadata) tuples.
+            if 'dict', the return value is a dictionary mapping {name: metadata}.
```

Comment:
I think these should say: a list/dict of tuples of the form: (system options, nonlinear solver options, linear solver options)   

above and below

I wonder if it would be more user-friendly if the dict format converted the tuples to dicts as well... e.g.
```
[
  { 'cycle':
       {'options':  { 'opt1': val1, 'opt2': val2},
       {'nonlinear': { 'opt1': val1, 'opt2': val2},
       {'linear': { 'opt1': val1, 'opt2': val2}
    },
    {'objcmp':
       {'options':  { 'opt1': val1, 'opt2': val2},
       {'nonlinear': { 'opt1': val1, 'opt2': val2},
       {'linear': { 'opt1': val1, 'opt2': val2}
    },
    etc..
 }    
```

Reply:
These are all good suggestions. I've pushed up a new version with them incorporated.